test/unit/test.discovery.v1.js

```diff
@@ -0,0 +1,177 @@
+    configurations: '/v1/environments/env-guid/configurations',
+    configurationinfo: '/v1/environments/env-guid/configurations/config-guid',
+    delete_collection: '/v1/environments/env-guid/collections/col-guid',
+    add_document: '/v1/environments/env-guid/collections/col-guid/documents',
+    delete_document: '/v1/environments/env-guid/collections/col-guid/documents/document-guid',
+    query: '/v1/environments/env-guid/collections/col-guid/query'
+  };
+
+  before(function() {
+    nock.disableNetConnect();
+    nock(service.url)
+      .persist()
+      .post(paths.environments + '?version=' + service.version_date)
+      .reply(200, {'environment_id': 'yes'})
+      .get(paths.environmentinfo + '?version=' + service.version_date)
+      .reply(200, {'environment_id': 'info'})
+      .get(paths.collections + '?version=' + service.version_date)
+      .reply(200, {'collection_id': 'yes'})
+      .get(paths.collectioninfo + '?version=' + service.version_date)
+      .reply(200, {'collection_id': 'info'})
+      .get(paths.query + '?version=' + service.version_date)
+      .reply(200, {'query': 'yes'})
+      .post(paths.collections + '?version=' + service.version_date)
+      .reply(200, {'collection_id': 'yes'})
+      .delete(paths.delete_collection + '?version=' + service.version_date)
+      .reply(200, {'config': 'yes'})
+      .post(paths.add_document + '?version=' + service.version_date)
+      .reply(200, {'add_doc': 'yes'})
+      .delete(paths.delete_document + '?version=' + service.version_date)
+      .reply(200, {'delete_doc': 'yes'})
+      .get(paths.configurations + '?version=' + service.version_date)
+      .reply(200, {'configs': 'yes'});
+  });
+
+  after(function() {
+    nock.cleanAll();
+  });
+
+  var missingParameter = function(err) {
+    assert.ok((err instanceof Error) && /required parameters/.test(err));
+  };
+
+  var discovery = new Discovery(service);
+
+  describe('discovery()', function() {
+    it('should generate a valid payload', function() {
+      var req = discovery.getEnvironments({}, noop);
+      assert.equal(req.uri.href, service.url + paths.environments + '?version=' + service.version_date);
+      assert.equal(req.method, 'GET');
+    });
+
+    it('should check no parameters provided (negative test)', function() {
+      discovery.getEnvironments({}, missingParameter);
+      discovery.getEnvironments(null, missingParameter);
+      discovery.getEnvironments(undefined, missingParameter);
+    });
+
+    it('should generate version_date was not specified (negative test)', function() {
+      var threw = false;
+      try {
+        // eslint-disable-next-line
+        var discovery = new Discovery(service1);
+      }
+      catch(err) {
+        threw = true;
+        assert.equal(err.message, 'Argument error: version_date was not specified, use DiscoveryV1.VERSION_DATE_2016_07_11');
+      }
+      assert(threw, 'should throw an error')
```

Comment:
BTW, there is an `assert.throws` method ;)

https://nodejs.org/api/assert.html#assert_assert_throws_block_error_message